src/Installer/Microsoft.Dotnet.Installation/Internal/DotnetArchiveExtractor.cs

```diff
@@ -56,68 +54,143 @@ public void Prepare()
         }
     }
     public void Commit()
-    {
-        Commit(GetExistingSdkVersions(_request.InstallRoot));
-    }
-
-    public void Commit(IEnumerable<ReleaseVersion> existingSdkVersions)
     {
         using var activity = InstallationActivitySource.ActivitySource.StartActivity("DotnetInstaller.Commit");
 
-
         using (var progressReporter = _progressTarget.CreateProgressReporter())
         {
             var installTask = progressReporter.AddTask($"Installing .NET SDK {_resolvedVersion}", maxValue: 100);
 
             // Extract archive directly to target directory with special handling for muxer
-            ExtractArchiveDirectlyToTarget(_archivePath!, _request.InstallRoot.Path!, existingSdkVersions, installTask);
+            ExtractArchiveDirectlyToTarget(_archivePath!, _request.InstallRoot.Path!, installTask);
             installTask.Value = installTask.MaxValue;
         }
     }
 
-    /**
-     * Extracts the archive directly to the target directory with special handling for muxer.
-     * Combines extraction and installation into a single operation.
-     */
-    private string? ExtractArchiveDirectlyToTarget(string archivePath, string targetDir, IEnumerable<ReleaseVersion> existingSdkVersions, IProgressTask? installTask)
+    /// <summary>
+    /// Extracts the archive directly to the target directory with special handling for muxer.
+    /// Combines extraction and installation into a single operation.
+    /// </summary>
+    private void ExtractArchiveDirectlyToTarget(string archivePath, string targetDir, IProgressTask? installTask)
     {
         Directory.CreateDirectory(targetDir);
 
-        var muxerConfig = ConfigureMuxerHandling(existingSdkVersions);
+        string muxerName = DotnetupUtilities.GetDotnetExeName();
+        string muxerTargetPath = Path.Combine(targetDir, muxerName);
+        string muxerTempPath = muxerTargetPath + ".tmp";
+
+        // Step 1: Read the version of the existing muxer (if any) by looking at the latest runtime
+        Version? existingMuxerVersion = null;
+        bool hadExistingMuxer = File.Exists(muxerTargetPath);
+        if (hadExistingMuxer)
+        {
+            existingMuxerVersion = GetLatestRuntimeVersionFromInstallRoot(targetDir);
+        }
 
+        // Step 2: If there is an existing muxer, rename it to .tmp
+        if (hadExistingMuxer)
+        {
+            try
+            {
+                if (File.Exists(muxerTempPath))
+                {
+                    File.Delete(muxerTempPath);
+                }
+                File.Move(muxerTargetPath, muxerTempPath);
+            }
+            catch
+            {
+                // If we can't rename, just continue - extraction will overwrite
```

Comment:
Let's change this to let it throw an exception.  It can throw an exception with a message and the inner exception for debugging.